scanpy/tests/test_highly_variable_genes.py

```diff
@@ -72,6 +72,25 @@ def test_highly_variable_genes_basic():
     assert np.all(np.isin(colnames, hvg_df.columns))
 
 
+@pytest.mark.parametrize('base', [None, 10])
+@pytest.mark.parametrize('flavor', ['seurat', 'cell_ranger'])
+def test_highly_variable_genes_keep_layer(base, flavor):
+    adata = pbmc3k()
+    # cell_ranger flavor can raise error if many 0 genes
+    sc.pp.filter_genes(adata, min_counts=1)
+
+    if flavor != 'seurat_v3':
```

Comment:
this condition is always true, since there’s a typo.

That’s why I always either include stuff like this in the params:

```py
@pytest.mark.parametrize(
    ('flavor', 'logarithmize', 'n_top_genes'),
    [
        pytest.param('seurat', True, 50, id='seurat'),
        pytest.param('cell_ranger', False, None, id='cell_ranger'),
    ],
)
```

or use a poor person’s [match statement](https://docs.python.org/3/reference/compound_stmts.html#grammar-token-python-grammar-match_stmt) (since we don’t have Python 3.10 as min version):

```py
if flavor == 'seurat':
    ...
elif flavor == 'cell_ranger':
    ...
else:
    assert False  # Completely fine in pytest code, not good in lib code, use `raise AssertionError('message')` there
```

Reply:
Wanted to test `seurat_v3` too in the beginning, but [this raises an error upon testing](https://dev.azure.com/scverse/scanpy/_build/results?buildId=4553&view=logs&j=50ff7263-9206-5a84-1219-938c9ee7fde7&t=2e49bd34-47bd-5a56-3183-6247e293d44d&l=1687) due to an additional dependency - indeed would have caught that with your suggestions, now cleaned up, thx!